src/pages/publication/references_tab/components/PublisherRow.tsx

```diff
@@ -50,9 +54,31 @@ interface PublisherRowProps {
 const PublisherRow: React.FC<PublisherRowProps> = ({ dataTestId, publisher, label, onClickDelete }) => {
   const { t } = useTranslation('publication');
 
+  const { setFieldValue }: FormikProps<FormikPublication> = useFormikContext();
+
   const { level, title } = publisher;
   const publisherLevel = typeof level === 'string' ? levelMap[level] : level;
 
+  useEffect(() => {
+    const setLevel = async (title: string) => {
+      const response = await getPublishers(title, PublicationTableNumber.PUBLICATION_CHANNELS);
+      // TODO: book = PublicationTableNumber.PUBLISHERS ?
+      if (response) {
+        const publisherLevel = response?.filter((publisher: Partial<Publisher>) => publisher.title === title)[0]?.level;
+        if (publisherLevel) {
+          const levelAsEnum = Object.keys(levelMap).find((key) => levelMap[key] === publisherLevel);
+          setFieldValue(`${ReferenceFieldNames.PUBLICATION_CONTEXT}.level`, levelAsEnum);
+        }
+      } else {
+        setFieldValue(`${ReferenceFieldNames.PUBLICATION_CONTEXT}.level`, '');
+      }
+    };
+
+    if (title && !level) {
+      setLevel(title);
+    }
+  }, [level, setFieldValue, title]);
```

Comment:
mistenker at denne vil gi en del ekstra unødvendige renders, men lever greit med det for nå.